### DISCUSSION TITLE
IP of the entity that does auth, IP of entity that submits request

### DISCUSSION THREAD

**Discussion by amypgon** [supabase/supabase#12705] [Feature Requests]
Hi there. When provisioning a database, could you please inject a function like auth.ip() (beside auth.uid, auth.role, auth.email) into the auth schema to achieve this: ip of the entity that does auth, ip of entity that submits request

Something along the following lines?

`current_setting('request.headers', true)::json->>'x-forwarded-for' `

This function could then be used when defining RLS policies

Thanks

**Comment by mansueli** [comment]
You probably want to take a look at this guide for hacking the postgREST headers:

https://dev.to/burggraf/hacking-the-postgrest-headers-oh-the-things-you-can-do-ck2

**Comment by amypgon** [comment]
Very interesting.
I see that it's possible to inject the functions via the SQL editor.
Is it possible to influence the db schema before Supabase has provisioned the db?
Thanks

**Comment by amypgon** [comment]
Here is an additional question. How to access or log all request headers directly in Supabase without using a client library? - https://github.com/supabase/supabase/discussions/12746
